### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by misterbisson**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM memcached:latest
+
+USER root
+# install python for entrypoint
+RUN apt-get update \
+    && apt-get install -y \
+		netcat \
+    curl
+
+COPY containerbuddy/* /opt/containerbuddy/
```
Comment (misterbisson) [inline Dockerfile]: See pattern demonstrated in https://github.com/autopilotpattern/jenkins:

Containerbuddy config in `/etc` and helper scripts in `/bin` (which is copied to `/usr/local/bin`) or just `/usr/local/bin` outright.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Install Containerbuddy
+# Releases at https://github.com/joyent/containerbuddy/releases
+ENV CONTAINERBUDDY_VERSION 0.1.1
```
Comment (misterbisson) [inline Dockerfile]: Use version 1.3; copy it from https://github.com/10up/triton-wordpress/blob/wordpress/Dockerfile#L46-L55

File: containerbuddy/config.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "consul": "consul:8500",
+  "services": [
+    {
+      "name": "memcached",
+      "port": 11211,
+      "publicIp": false,
```
Comment (misterbisson) [inline containerbuddy/config.json]: Specify interface priority, per https://github.com/autopilotpattern/wordpress/pull/3/files#r58486341

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ENV CONTAINERBUDDY_SHA1 3163e89d4c95b464b174ba31733946ca247e068e
+RUN curl --retry 7 -Lso /tmp/containerbuddy.tar.gz "https://github.com/joyent/containerbuddy/releases/download/${CONTAINERBUDDY_VERSION}/containerbuddy-${CONTAINERBUDDY_VERSION}.tar.gz" \
+    && echo "${CONTAINERBUDDY_SHA1}  /tmp/containerbuddy.tar.gz" | sha1sum -c \
+    && tar zxf /tmp/containerbuddy.tar.gz -C /opt/containerbuddy \
+    && rm /tmp/containerbuddy.tar.gz
+
+
+
+COPY docker-entrypoint.sh /entrypoint.sh
+ENTRYPOINT ["/entrypoint.sh"]
```
Comment (misterbisson) [inline Dockerfile]: The use of this script will trap signals and they won't get to Containerbuddy, which will break some features. Better perhaps just to remove it.

File: docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+# first check if we're passing flags, if so
+# prepend with memcached
+if [ "${1:0:1}" = '-' ]; then
+	set -- memcached "$@"
```
Comment (misterbisson) [inline docker-entrypoint.sh]: This entrypoint script allows starting the image with args like this:

`docker run memached -l 0.0.0.0`

Without it, we can still pass args, like this:

`docker run memached memcached -l 0.0.0.0`

Given https://github.com/autopilotpattern/memcached/pull/1#discussion_r58487244, I'd get rid of it.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ENV CONTAINERBUDDY file:///etc/containerbuddy.json
+RUN export CONTAINERBUDDY_CHECKSUM=c25d3af30a822f7178b671007dcd013998d9fae1 \
+    && curl -Lso /tmp/containerbuddy.tar.gz \
+         "https://github.com/joyent/containerbuddy/releases/download/${CONTAINERBUDDY_VER}/containerbuddy-${CONTAINERBUDDY_VER}.tar.gz" \
+    && echo "${CONTAINERBUDDY_CHECKSUM}  /tmp/containerbuddy.tar.gz" | sha1sum -c \
+    && tar zxf /tmp/containerbuddy.tar.gz -C /usr/local/bin \
+    && rm /tmp/containerbuddy.tar.gz
+
+USER memcache
+CMD ["/opt/containerbuddy/containerbuddy", \
```
Comment (misterbisson) [inline Dockerfile]: This is solid. My only question is if the `ENTRYPOINT` defined in the base image is being executed or not. 

Actually, that question is relevant to https://github.com/autopilotpattern/wordpress/pull/1 as well.

Comment (jasondewitt) [inline Dockerfile]: It looks like it was being run, based on this showing up in the 'docker inspect' output:

```
        "Entrypoint": [
            "/entrypoint.sh"
        ],
```

I found this https://github.com/docker/docker/issues/3465#issuecomment-47596890
which says you can reset the entrypoint of a parent image. I tested that here on the memcached image and the entrypoint stansa of 'docker inspect' no longer shows up.

Will push up this change shortly.

Comment (misterbisson) [inline Dockerfile]: Thanks for confirming that. It's a frustrating enough problem that you should probably add an inline comment in the Dockerfile explaining it, if you haven't already planned to.

**Comment by misterbisson** [comment]
@roXet if this is working for you (with my changes last week), let's merge it to master

**Comment by jasondewitt** [comment]
adding a README.md, there's not much that needs to be done to get this up and running. Do we think this simple readme covers it well enough? Got the badges in there, that should work after we push the image to Docker Hub.

Everything in here works well for me in my testing, I think we can merge to master.

**Pull request merged**